System.Containers.Tasks/CreateNewImage.cs

```diff
@@ -1,105 +1,146 @@
-        /// </summary>
-        [Required]
-        public string BaseImageName { get; set; }
-
-        [Required]
-        public string BaseImageTag { get; set; }
-
-        [Required]
-        public string InputRegistryURL { get; set; }
-
-        [Required]
-        public string OutputRegistryURL { get; set; }
-
-        /// <summary>
-        /// Constructed from "$(MSBuildProjectDirectory)\$(PublishDir)"
-        /// </summary>
-        [Required]
-        public string PublishDirectory { get; set; }
-
-        /// <summary>
-        /// $(ContainerWorkingDirectory)
-        /// </summary>
-        [Required]
-        public string WorkingDirectory { get; set; }
-
-        [Required]
-        public string NewImageName { get; set; }
-
-        [Required]
-        public string Entrypoint { get; set; }
-
-        /// <summary>
-        /// Arguments to pass alongside Entrypoint.
-        /// </summary>
-        public string EntrypointArgs { get; set; }
-
-
-
-        /// <summary>
-        /// CreateNewImage needs to:
-        /// 1. Pull a base image (needs parameters: URL, BaseImage, BaseImageTag)
-        /// 2. Add output of build as a new layer
-        /// 3. Push image back to some registry (needs parameters: OutputURL, NewName, EntryPoint)
-        /// </summary>
-        /// <returns></returns>
-        public override bool Execute()
+        BaseRegistry = "";
+        BaseImageName = "";
+        BaseImageTag = "";
+        OutputRegistry = "";
+        ImageName = "";
+        ImageTag = "";
+        PublishDirectory = "";
+        WorkingDirectory = "";
+        Entrypoint = "";
+        EntrypointArgs = "";
+    }
+
+
+    public override bool Execute()
+    {
+        if (string.IsNullOrEmpty(PublishDirectory))
         {
-            if (string.IsNullOrEmpty(PublishDirectory) || !Directory.Exists(PublishDirectory))
-            {
-                Log.LogError("PublishDirectory and Files are both invalid. One valid parameter MUST be given to the CreateNewImage task.");
-                return false;
-            }
+            Log.LogError(string.Format("PublishDirectory is must have a value"));
+            return !Log.HasLoggedErrors;
+        }
+        else if (!Directory.Exists(PublishDirectory))
+        {
+            Log.LogError(string.Format("PublishDirectory does not exist."));
+            return !Log.HasLoggedErrors;
+        }
 
-            Registry reg = new Registry(new Uri(InputRegistryURL));
+        Registry reg;
+        Image image;
 
-            Image image;
-            try
-            {
-                image = reg.GetImageManifest(BaseImageName, BaseImageTag).Result;
-            }
-            catch (Exception ex)
+        try
+        {
+            reg = new Registry(new Uri(BaseRegistry, UriKind.RelativeOrAbsolute));
+            image = reg.GetImageManifest(BaseImageName, BaseImageTag).Result;
+        }
+        catch (Exception e)
+        {
+            if (BuildEngine != null)
```

Comment:
I'm assuming the BuildEngine checks are because there's no easily-mockable build engine? Where's that MSBuild issue to ship something out of box.... :D

Reply:
I briefly chatted with rainer about there being no built in mockengine. +10 to building one in!